internal/models/reservation_model.go

```diff
@@ -35,6 +45,20 @@ type NoopReservation struct {
 	Reservation
 }
 
+type AWSDetail struct {
+	// Optional instance name
+	Names *[]string `json:"names"`
+
+	// AWS Instance type.
+	InstanceType string `json:"instance_type"`
+
+	// Amount of instances to provision of type: Instance type.
+	Amount int32 `json:"amount"`
+
+	// Immediately power off the system after initialization
+	PowerOff bool `json:"poweroff"`
+}
+
```

Comment:
This is a new type that is serialized into database as JSONB. It is embedded into the reservation type under name `Detail`.